base/src/MultimediaQueue.cpp

```diff
@@ -0,0 +1,789 @@
+#include <boost/filesystem.hpp>
+#include <boost/foreach.hpp>
+#include <stdafx.h>
+#include <map>
+#include "Frame.h"
+#include "MultimediaQueue.h"
+#include "Logger.h"
+#include "H264Utils.h"
+#include "EncodedImageMetadata.h"
+#include "H264Metadata.h"
+
+class QueueClass
+{
+public:
+	typedef std::map<uint64_t, frame_container> MultimediaQueueMap;
+	MultimediaQueueMap mQueue;
+	virtual bool enqueue(frame_container& frames, uint32_t lowerWaterMark, uint32_t upperWaterMark, bool isMapDelayInTime, bool  pushNext) { return true; }
+	const_buffer spsBuffer;
+	const_buffer ppsBuffer;
+};
+
+class JpegQueueClass : public QueueClass
+{
+
+public:
+	~JpegQueueClass()
+	{}
+
+	bool enqueue(frame_container& frames, uint32_t lowerWaterMark, uint32_t upperWaterMark, bool isMapDelayInTime, bool  pushNext)
+	{	//	Here the frame_containers are inserted into the map
+		uint64_t largestTimeStamp = 0;
+		for (auto it = frames.cbegin(); it != frames.cend(); it++)
+		{
+			mQueue.insert({ it->second->timestamp, frames });
+			if (largestTimeStamp < it->second->timestamp)
+			{
+				largestTimeStamp = it->second->timestamp;
+			}
+		}
+
+		if (isMapDelayInTime) // If the lower and upper watermark are given in time
+		{
+			if ((largestTimeStamp - mQueue.begin()->first > lowerWaterMark) && (pushNext == true))
```

Comment:
a boolean can be tested as (...&& pushNext ) there is no value in using an eqality opertaor